src/app/rpmostree-builtin-kargs.c

```diff
@@ -62,6 +59,104 @@ get_kargs_option_variant (void)
+ * inside the string gets swapped out to empty. Then, the
+ * other terms inside the kernel args are getting compared
+ */
+static gboolean
+kernel_arg_string_equal (const char* a_kernel_arg_string,
+                         const char* b_kernel_arg_string)
+{
+  __attribute__((cleanup(_ostree_kernel_args_cleanup))) OstreeKernelArgs *a_kargs = _ostree_kernel_args_from_string (a_kernel_arg_string);
+  _ostree_kernel_args_replace (a_kargs, "ostree");
+  g_autofree char* a_kernel_arg_string_without_ostree = _ostree_kernel_args_to_string (a_kargs);
+
+  __attribute__((cleanup(_ostree_kernel_args_cleanup))) OstreeKernelArgs *b_kargs = _ostree_kernel_args_from_string (b_kernel_arg_string);
+  _ostree_kernel_args_replace (b_kargs, "ostree");
+  g_autofree char* b_kernel_arg_string_without_ostree = _ostree_kernel_args_to_string (b_kargs);
+  if (strcmp (a_kernel_arg_string_without_ostree, b_kernel_arg_string_without_ostree) != 0)
+    return FALSE;
+  return TRUE;
+}
+
+/*
+ * This function shares a similar logic as the commit_editor
+ * function in ostree code base. It takes old kernel arguments
+ * and display to user through an editor. Then, then user
+ * is able to modify those args.
+ */
+static gboolean
+kernel_arg_handle_editor (const char     *input_kernel_arg,
+                          const char    **out_kernel_arg,
+                          GCancellable   *cancellable,
+                          GError        **error)
+{
+  g_autofree char *chomped_input = g_strchomp (g_strdup (input_kernel_arg));
+  g_autofree char *input_string = g_strdup_printf ("\n"
+      "# Please enter the kernel arguments. Each kernel argument"
+      "# should be in the form of key=value.\n"
+      "# Lines starting with '#' will be ignored. Each key=value pair should be \n"
+      "# separated by spaces, and multiple value associated with one key is allowed. \n"
+      "# Also, please note that any changes to existing ostree argument will not be \n"
+      "# effective as they are usually regenerated when bootconfig changes \n"
+      "%s\n", chomped_input);
+
+  /* Note: the repo here is NULL, as we don't really require it
+   * for the edtior process
+   */
+  g_autofree char *out_editor_string = ot_editor_prompt (NULL, input_string,
+                                                         cancellable, error);
+  if (out_editor_string == NULL)
+    return FALSE;
+
+  /* We now process the editor output, we ignore lines starting with #
+   * and we assume the kernel arguments are usually in one line.
+   */
+  g_auto(GStrv) lines = g_strsplit(out_editor_string, "\n", -1);
+  guint num_lines = g_strv_length (lines);
+  g_autoptr(GString) kernel_arg_buf = g_string_new ("");
+  for (guint i = 0; i < num_lines; i++)
+    {
+      /* remove leading and trailing spaces */
+      g_strchomp (lines[i]);
+      g_strchug (lines[i]);
+      /* comments and blank lines are skipped */
+      if (lines[i][0] == '#')
```

Comment:
This would read a bit more nicely I think if we did `char *line = lines[i];` at the top, then `*line == '#'` or so.